build/org.eclipse.cdt.managedbuilder.core.tests/tests/org/eclipse/cdt/managedbuilder/core/tests/CompilationDatabaseGenerationTest.java

```diff
@@ -167,4 +171,34 @@ public void restoreDefaultForGenerateFile() {
 				"org.eclipse.cdt.managedbuilder.ui");
 		preferenceStore.setToDefault(CommonBuilder.COMPILATION_DATABASE_ENABLEMENT);
 	}
+
+	@Test
+	public void testCompilerPath() throws Exception {
+		setWorkspace("regressions");
+		final IProject app = loadProject("helloworldC");
+		setGenerateFileOptionEnabled(true);
+		app.build(IncrementalProjectBuilder.FULL_BUILD, null);
+		IFile commandsFile = app.getFile("Debug/compile_commands.json");
+
+		if (commandsFile.exists()) {
+			try (FileReader reader = new FileReader(commandsFile.getLocation().toFile())) {
+				Gson gson = new Gson();
+				JsonArray jsonArray = gson.fromJson(reader, JsonArray.class);
+				for (JsonElement element : jsonArray) {
+					CompilationDatabaseInformation compileCommand = gson.fromJson(element,
+							CompilationDatabaseInformation.class);
+					String command = compileCommand.command();
+					String[] commandParts = CommandLineUtil.argumentsToArray(command);
+					String compilerPath = commandParts[0];
+					assertNotNull("Compiler path should not be null", compilerPath);
+					assertFalse("Compiler path should not be empty", compilerPath.isEmpty());
+					IPath path = new Path(compilerPath);
+					//Added this to see the source of the error and try to fix
```

Comment:
This code looks correct - I hope it is enough to diagnose the build machine error. 

Please tag me once you need my input again.

Reply:
Hello @jonahgraham , your suggestion to use CommandLineUtil.argumentsToArray fixed the issue because the method will correctly treat the quoted string as a single argument, no matter the OS. The way I was splitting the command before was the problem with the Linux environment. 

Do you think I need to add anything else?

Reply:
Just to remove the added logging/debugging code. I didn't look thoroughly, but did mark one instance in latest review.